### PR TITLE
Restrict external forms to a single project, check can_manage_external_form_submissions against project entity

### PR DESCRIPTION
**Pull request by gigxz**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

* Add rails validation to ensure that External Form can only have 1 active rule, and that rule must be project-level
* When checking `can_manage_external_form_submissions` perm, check against the project that actively "reviews" the form submission

Issue: https://github.com/open-path/Green-River/issues/6628
Frontend PR: https://github.com/greenriver/hmis-frontend/pull/923

## Type of change
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: drivers/hmis/app/graphql/mutations/create_form_rule.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class CreateFormRule < BaseMutation
     argument :definition_id, ID, required: true
     argument :input, Types::Admin::FormRuleInput, required: true
 
-    field :form_rule, Types::Admin::FormRule, null: false
+    field :form_rule, Types::Admin::FormRule, null: true
```
Comment (gigxz) [inline drivers/hmis/app/graphql/mutations/create_form_rule.rb]: moved these changes to https://github.com/greenriver/hmis-warehouse/pull/4703

File: drivers/hmis/app/graphql/mutations/update_external_form_submission.rb [hunk 2]
```diff
@@ -8,14 +8,16 @@ module Mutations
   class UpdateExternalFormSubmission < CleanBaseMutation
     argument :input, Types::HmisSchema::ExternalFormSubmissionInput, required: true
     argument :id, ID, required: true
-    argument :project_id, ID, required: false # not required for backwards compatibility
```
Comment (gigxz) [inline drivers/hmis/app/graphql/mutations/update_external_form_submission.rb:11]: This is just walking back a change that is only on `pit-external-forms` https://github.com/greenriver/hmis-warehouse/compare/release-133...pit-external-forms?expand=1#diff-d340d41adf901f7723f0a39b77b876ff73100e13027c74cb5a7a666b12d7a2bd

File: drivers/hmis/app/graphql/types/hmis_schema/project.rb [hunk 3]
```diff
@@ -229,9 +229,10 @@ def outgoing_referral_postings(**args)
     end
 
     def external_form_submissions(**args)
-      # Not optimized performance, but reduces duplication of project_match logic
-      instances = Hmis::Form::Instance.with_role(:EXTERNAL_FORM) # Don't filter by active; show all past submissions
-      identifiers = instances.for_project_through_entities(object).select(:definition_identifier)
+      # Find form instances for this project. Only include active instances. (Unlike other form types, we do not support
+      # viewing "legacy" form submissions from inactive instances, to simplify implementation.)
+      instances = Hmis::Form::Instance.with_role(:EXTERNAL_FORM).for_project(object).active
```
Comment (martha) [inline drivers/hmis/app/graphql/types/hmis_schema/project.rb:235]: Perhaps add comment here about why we don't use `for_project_through_entities` (similar to pick_list_option above)

**Review by martha — APPROVED** [review]
Wonderful, thanks!

**Pull request merged**